llama/model_cn_comment.py

```diff
@@ -86,50 +86,73 @@ def forward(self, x):
 
 
 def precompute_freqs_cis(dim: int, end: int, theta: float = 10000.0):
-    """Precompute the frequency tensor for complex exponentials (cis) with given dimensions.
-
-    This function calculates a frequency tensor with complex exponentials using the given dimension
-    'dim' and the end index 'end'. The 'theta' parameter scales the frequencies.
-    The returned tensor contains complex values in complex64 data type.
+    """根据指定维度预计算复指数 (cis) 的频率张量. 
+    
+    cis 是一种用于缩略表示欧拉公式的数学标记: cis x = cos x + i sin x.
+    该函数根据给定的维度 dim 和结束索引 end, 计算一个复指数的频率张量.
+    theta 参数用于缩放频率.
+    返回的张量包含复数值，数据类型为 complex64.
 
     Args:
-        dim (int): Dimension of the frequency tensor.
-        end (int): End index for precomputing frequencies.
-        theta (float, optional): Scaling factor for frequency computation. Defaults to 10000.0.
+        dim (int): 频率张量的维度, 也就是 self-attention 中 query 和 key 向量的维度.
+        end (int): 预计算频率的结束索引, 也就是模型支持的最长序列长度.
+        theta (float, optional): 频率计算的缩放因子, 默认为 10000.0.
 
     Returns:
-        torch.Tensor: Precomputed frequency tensor with complex exponentials.
+        torch.Tensor: 预计算的复指数的频率张量. Shape: (end, dim / 2).
     """
+    # 根据 theta 和 dim 参数计算用于 RoPE 不同维度的旋转角度的基.
+    # 即计算 θ_i = theta^{-2i / dim}, i ∈ [0, dim / 2).
+    # [Shape] freqs: (dim / 2, )
     freqs = 1.0 / (
         theta ** (torch.arange(0, dim, 2)[: (dim // 2)].float() / dim)
     )
-    t = torch.arange(end, device=freqs.device)  # type: ignore
-    freqs = torch.outer(t, freqs).float()  # type: ignore
+    
+    # 根据结束索引 end 生成用于采样所有位置旋转角度的 t.
+    # [Shape] t: (end, )
+    t = torch.arange(end, device=freqs.device)
```

Comment:
`theta` 会不会比 `t` 更 formal 一些呢？感觉单独一个 `t` 实在是过于简洁了。
  
```suggestion
    theta = torch.arange(end, device=freqs.device)
```

Reply:
才发现上面已经有一个 theta 作为参数了。但是我认为单独一个 t 会让读者读起来很 confused。

Reply:
目前我的处理逻辑是不更改源代码只做 comment。

Reply:
Get